app_flutter/lib/state/flutter_build.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+import 'package:cocoon_service/protos.dart' show CommitStatus;
+
+import '../service/cocoon.dart';
+
+/// State for the Flutter Build Dashboard
+class FlutterBuildState extends ChangeNotifier {
+  /// Cocoon backend service that retrieves the data needed for this state.
+  final CocoonService _cocoonService = CocoonService();
+
+  /// How often to query the Cocoon backend for the current build state.
+  final Duration _refreshRate = Duration(seconds: 10);
+
+  /// Timer that calls [_fetchBuildStatusUpdate] on a set interval.
+  Timer _refreshTimer;
+
+  /// The current status of the commits loaded.
+  List<CommitStatus> statuses = [];
+
+  /// Start a fixed interval loop that fetches build state updates based on [_refreshRate].
+  void startFetchingBuildStateUpdates() async {
+    if (_refreshTimer != null) {
+      throw 'already fetching build state updates';
```

Comment:
While it's not a compile error to throw a String, generally you should only throw `Exception` or `Error`. That said, I don't think this should throw at all, just return if it is already set.

Reply:
Done. I modified the test to check if the `refreshTimer` is the same across multiple start calls.

Thanks for sharing the use about `Exception` and `Error`. I saw the backend doing it that way so I assumed that was the better practice :)